pilot/pkg/model/push_context.go

```diff
@@ -2173,17 +2174,11 @@ func (ps *PushContext) ServiceAccounts(hostname host.Name, namespace string, por
 	}]
 }
 
-func (ps *PushContext) SupportsTunnel(ip string) bool {
-	infos, _ := ps.ambientIndex.AddressInformation(sets.New(types.NamespacedName{Name: ip}))
-	for _, p := range infos {
-		switch addr := p.Address.Type.(type) {
-		case *workloadapi.Address_Workload:
-			if addr.Workload.TunnelProtocol == workloadapi.TunnelProtocol_HBONE {
-				return true
-			}
-		case *workloadapi.Address_Service:
-			// Services do not directly support tunneling, their individual Workloads can though
-			continue
+func (ps *PushContext) SupportsTunnel(n network.ID, ip string) bool {
+	infos, _ := ps.ambientIndex.AddressInformation(sets.New(n.String() + "/" + ip))
```

Comment:
I don't mind using string instead of NamespacedName - but isn't it a bit inconsistent with the rest of the code ? 

Also a bit theoretical - isn't string concatenation more expensive than using a stuct ? Doesn't matter here, small strings and not in the critical path, for my own curiosity.

Reply:
+1